dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/sms/ProgramStageDataEntrySMSListener.java

```diff
@@ -187,20 +231,16 @@ private boolean hasMoreThanOneEntity(List<TrackedEntity> trackedEntities) {
   }
 
   private TrackedEntityOperationParams getParams(
-      TrackedEntityAttribute attribute,
-      IncomingSms sms,
-      Program program,
-      Set<OrganisationUnit> ous) {
+      TrackedEntityAttribute attribute, IncomingSms sms, Program program) {
 
     QueryFilter queryFilter = new QueryFilter();
-    queryFilter.setOperator(QueryOperator.LIKE);
+    queryFilter.setOperator(QueryOperator.EQ);
     queryFilter.setFilter(sms.getOriginator());
 
     return TrackedEntityOperationParams.builder()
         .filters(Map.of(attribute.getUid(), List.of(queryFilter)))
-        .programUid(program.getUid())
-        .organisationUnits(
-            ous.stream().map(BaseIdentifiableObject::getUid).collect(Collectors.toSet()))
+        .trackedEntityTypeUid(program.getTrackedEntityType().getUid())
```

Comment:
If the phone number we use to search for a TE, can it be either a program TEAttribute or a TETypeAttribute?  In cases where the phone number is a program tea, does this filter on TET here block us from finding the TE with the phone number?  Probably not. Just curious.

Reply:
This SMS can also enroll a TE. We can not find the TE if we query using the program 😅